### PR TITLE
Apply hljs classPrefix to container figure element

### PR DESCRIPTION
**Pull request by Krakob**
Building on #15. As hljs uses 'hljs' by default for its containers, and not 'highlight' as hexo currently has hardcoded, this makes sure that the original hljs stylesheets maintain compatibility, should you use hljs as your parameter.

It should be noted that a `-` is appended to the class prefix so that we get, for example, `hljs` for the container and `hljs-number`. Also added some quick documentation as was requested in #15.

### CODE REVIEW

**Comment by Krakob** [comment]
Not quite sure what's up with the test, but it seems to try to close some spans twice, e.g.

`a = <span class="hljs-string">`</span><br>` expected,
`a = <span class="hljs-string">`</span></span><br>` received

Can't say I know unit testing very well.

**Comment by NoahDragon** [comment]
I will have a look.

File: lib/highlight.js [hunk 1]
```diff
@@ -5,14 +5,21 @@ var Entities = require('html-entities').XmlEntities;
 var entities = new Entities();
 var alias = require('../highlight_alias.json');
 
-hljs.configure({
-  classPrefix: ''
-});
+// hljs.configure({
+//   classPrefix: ''
+// });
```
Comment (NoahDragon) [inline lib/highlight.js]: Could you please remove the comment out codes?

Comment (Krakob) [reply lib/highlight.js]: Done :)

File: lib/highlight.js [hunk 2]
```diff
@@ -5,14 +5,17 @@ var Entities = require('html-entities').XmlEntities;
...
-hljs.configure({
-  classPrefix: ''
-});
-
 function highlightUtil(str, options) {
   if (typeof str !== 'string') throw new TypeError('str must be a string!');
   options = options || {};
 
+  var classPrefix = options.hasOwnProperty('classPrefix') ? options.classPrefix + '-' : '';
+  var containerPrefix = options.hasOwnProperty('classPrefix') ? options.classPrefix : 'highlight';
```
Comment (NoahDragon) [inline lib/highlight.js]: Shouldn't we remove this line?

**Comment by JLHwung** [comment]
The unit-test failed on `classPrefix`, could you give it a fix?

**Comment by Krakob** [comment]
I can't figure out how :(

The feature was working in the initial PR, as seen here: https://travis-ci.org/hexojs/hexo-util/jobs/146592708

Yet this commit failed: https://github.com/hexojs/hexo-util/pull/29/commits/909523ad3e198c24cc7ff9a572ce7fdec02e775d

And the failure reason being, as mentioned, generating duplicate closing tags for some reason, which sounds like an internal issue in hljs. The only line in the lib file we can link directly to hljs is 16, but it's compensated for in the test as the hyphen in classPrefix was removed.

**Comment by NoahDragon** [comment]
@Krakob The config for classPrefix in highlight.js seems changed the behavior of generating lines.

Expected a line:
```html
<span class="line">
    <span class="hljs-string">  Multi</span>
</span>
<br>
```

The line generated via higlight.js with the classPrefix:
```html
<span class="line">  Multi</span>
<br>
```

Not sure how this happens, will investigate deeper.

**Comment by JLHwung** [comment]
closed in favor of #30.

**Pull request closed without merging**